internal/lightcone/abundance/quidproquo/quidproquo.go

```diff
@@ -45,9 +45,7 @@ func randomlyAddEnergy(mod *modifier.ModifierInstance) {
 	
 	for _, char := range allyList{
 		//check if energy is <50% and current char isn't LC's holder.
-		//NOTE : Engine().CharacterInfo() don't have .MaxEnergy(). 
-		//currently using .Stats which is not efficient.
-		if (mod.Engine().Energy(char) < mod.Engine().Stats(char).MaxEnergy() / 2 && 
+		if (mod.Engine().Energy(char) < mod.Engine().MaxEnergy(char) / 2 && 
```

Comment:
I suspect in the future we may get other checks like this, but I doubt anything would ever specifically care about the `MaxEnergy` of a character (because that is something fixed and should never change). Might be better to just expose an `EnergyRatio` method instead similar to `HPRatio` we currently have, that way this simplifies to just:
```go
if mod.Engine().EnergyRatio(char) < 0.5 && char != mod.Owner() {
  ... 
}
```